### PR TITLE
feat(schemas): [OWL-27002516] Implemented new data model for users

### PR DESCRIPTION
**Pull request by devthena**
Notes:
- The `/star` command is depending on the new activity tracking collection for the `last_star` value
- All commands both on Discord and Twitch need to be tested before this gets merged

### CODE REVIEW

File: src/discord/commands/gamble.ts [hunk 1]
```diff
@@ -73,62 +73,56 @@ export const Gamble = {
...
-        ?.collection(process.env.MONGODB_USERS)
-        .updateOne(
-          { discord_id: user.discord_id },
-          { $set: { ...updates } },
-          { upsert: true }
-        );
-    }
+    await Bots.db
+      ?.collection(Bots.env.MONGODB_USERS)
+      .updateOne({ discord_id: user.discord_id }, { $set: { cash: points } });
```
Comment (alvinb) [inline src/discord/commands/gamble.ts:125]: would this throw an error if `Bots.env.MONGODB_USERS`  is not defined?

File: src/discord/events/interactionCreate.ts [hunk 2]
```diff
@@ -20,31 +22,46 @@ export const onInteractionCreate = async (
...
-    const data: DiscordUserProps = {
-      discord_id: interaction.member.user.id,
-      discord_name: interaction.member.user.username,
-      discord_tag: document?.discord_tag || discordTag,
-      last_star: document?.last_star || '',
-      points: document?.points || 0,
-      stars: document?.stars || 0,
-    };
+    if (!document) {
+      await Bots.db?.collection(Bots.env.MONGODB_USERS).insertOne(userData);
```
Comment (alvinb) [inline src/discord/events/interactionCreate.ts:58]: i think we might need to test this if the insert fails somehow. 
im thinking of creating a new ticket for error handling.

File: src/index.ts [hunk 3]
```diff
@@ -50,22 +50,18 @@ const Bots: BotsProps = {
...
     ],
   }),
   env: {
     ADMIN_SERVER_ID: process.env.ADMIN_SERVER_ID || '',
-    DISCORD_CLIENT_ID: process.env.DISCORD_CLIENT_ID || '',
-    DISCORD_TOKEN: process.env.DISCORD_TOKEN || '',
-    MONGODB_USERS: process.env.MONGODB_USERS || '',
-    MONGODB_CHAT: process.env.MONGODB_CHAT || '',
-    MONGODB_VIEW: process.env.MONGODB_VIEW || '',
+    MONGODB_USERS: 'users',
```
Comment (alvinb) [inline src/index.ts:64]: i see. so this was moved out from the .env file. ignore my previous comment at the top :)

File: src/discord/events/interactionCreate.ts (hunk 2 above)
Comment (devthena) [reply src/discord/events/interactionCreate.ts:58]: I think a good practice is adding the `try` and `catch` statements when making async/await statements. Let's do that on the ticket you created for error handling.

Comment (devthena) [reply src/discord/events/interactionCreate.ts:58]: I'm also gonna add some @todo comments on these so we can easily reference it later.

**Review by alvinb — APPROVED** [review]
approving but from your comments we need to test it first?
I can help test so just let me know

**Pull request merged**